keras_core/backend/torch/core.py

```diff
@@ -170,7 +160,10 @@ def cast(x, dtype):
     if isinstance(x, KerasVariable):
         x = x.value
     if is_tensor(x):
-        return x.to(dtype)
+        if x.dtype == dtype:
+            return x
+        else:
+            return x.to(dtype)
```

Comment:
Why are we putting an overhead of if-else here?

Reply:
It turns out `x.to` has more overhead. This would save some time by only call it when necessary.